tt_metal/common/blockfloat_common.hpp

```diff
@@ -284,8 +284,18 @@ inline std::vector<uint32_t> pack_fp32_vec_as_bfp_tiles(const std::vector<float>
         BfpFormat == tt::DataFormat::Bfp8_b
     );
 
-    uint32_t single_bfp8_tile_size = tile_size(BfpFormat);
-    int num_float_in_tile = 1024;
+    auto tile_H = tile.has_value() ? tile->get_tile_shape()[0] : tt::constants::TILE_HEIGHT;
+    auto tile_W = tile.has_value() ? tile->get_tile_shape()[1] : tt::constants::TILE_WIDTH;
+    auto face_H = tile.has_value() ? tile->get_face_shape()[0] : tt::constants::FACE_HEIGHT;
+    auto face_W = tile.has_value() ? tile->get_face_shape()[1] : tt::constants::FACE_WIDTH;
```

Comment:
There is way too much duplicate code.

`tile` should not be optional in any method. Instead the default parameter should be a default Tile constant defined in tt::constants

Reply:
Will this request be addressed? Or not planned anymore?